### PR TITLE
Environment configuration externalization

### PR DESCRIPTION
**Pull request by Nigui**
This PR initialize a solution to externalize app configuration, depending on target environment.

Solution chooses if npm module [`config`](https://www.npmjs.com/package/config)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,19 +9,20 @@
   "dependencies": {
     "@types/body-parser": "^1.16.8",
     "@types/express": "^4.11.1",
+    "config": "^1.30.0",
+    "trackr-lib": "sonarplanet/trackr-lib#publish",
     "web-push": "^3.3.0",
-    "trackr-lib": "sonarplanet/trackr-lib#publish"
+    "typescript": "^2.8.1"
```
Comment (dlecan) [inline package.json]: Do we really need Typescript at runtime?

File: config/default.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (dlecan) [inline config/default.json]: I would recommend to use typescript config file format.
https://github.com/lorenwest/node-config/wiki/Configuration-Files#typescript---ts

We can define a common interface, default values, which must be overloaded in each environment configuration.

File: src/app.ts [hunk 3]
```diff
@@ -1,65 +1,76 @@
...
-} from 'body-parser'
-import * as webpush from 'web-push'
-import * as trackr from 'trackr-lib'
+import * as express from "express";
+import { json } from "body-parser";
+import * as webpush from "web-push";
+import * as trackr from "trackr-lib";
 
-const app = express()
+const CONFIG = require("config");
```
Comment (dlecan) [inline src/app.ts]: `import`!

File: package.json [hunk 4]
```diff
@@ -9,19 +9,20 @@
   "dependencies": {
     "@types/body-parser": "^1.16.8",
     "@types/express": "^4.11.1",
```
Comment (dlecan) [inline package.json]: Missing import: `@types/config`

File: config/production.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (dlecan) [inline config/production.json]: I would recommend empty files for the other environments: integ & staging

**Review by dlecan — DISMISSED** [review]
Good work!

I would have been great to format all files in the project before doing this work. Real changes are hard to see.

**Review by ftreguer — APPROVED** [review]

**Pull request merged**